addons/account/models/account_journal.py

```diff
@@ -540,7 +540,10 @@ def write(self, vals):
                     if bank_account.partner_id != company.partner_id:
                         raise UserError(_("The partners of the journal's company and the related bank account mismatch."))
             if 'restrict_mode_hash_table' in vals and not vals.get('restrict_mode_hash_table'):
-                journal_entry = self.env['account.move'].sudo().search([('journal_id', '=', self.id), ('state', '=', 'posted'), ('secure_sequence_number', '!=', 0)], limit=1)
+                domain = self.env['account.move']._get_move_hash_domain(
+                    common_domain=[('journal_id', '=', self.id), ('secure_sequence_number', '!=', 0)]
+                )
+                journal_entry = self.env['account.move'].sudo().search(domain, limit=1)
```

Comment:
Not sure why we would need the `_get_move_hash_domain`.
I think the `common_domain` is enough on its own :thinking:

Reply:
I think we still need it because we're looking for moves that are restricted. Previously, we had `('state', '=', 'posted')` in the domain and I had to remove that part, replacing with `_get_move_hash_domain`